src/com/esotericsoftware/kryo/serializers/AsmField.java

```diff
@@ -23,7 +23,7 @@
 import com.esotericsoftware.kryo.io.Input;
 import com.esotericsoftware.kryo.io.Output;
 import com.esotericsoftware.kryo.serializers.FieldSerializer.CachedField;
-import com.esotericsoftware.kryo.util.Generics.GenericType;
+import com.esotericsoftware.kryo.util.DefaultGenericsStrategy.GenericType;
```

Comment:
Not sure about this: does it make sense, that `GenericType` is defined within `DefaultGenericsStrategy` (and not within `GenericsStrategy`)? On the one hand I think yes, one the other hand I wonder why some classes should refer to s.th. that's closely related to a specific implementation...

Reply:
That was the result of renaming the old Generics class. As you said, it does not make much sense for GenericsType and GenericsHierarchy to reside in the DefaultGenericsStrategy.
Note that since it is still be accessible via the GenericsStrategy, it is still part of the public interface (which it also should be, given that it was already be usable from users of this library).
Instead of having them as static inner classes, I've moved them to their own files. I personally think this is cleaner, but we probably could create them as inner classes of the interface as well. They seem to be data classes and implementation independent as well, so I think it should not be necessary to abstract them using interfaces as well?